File: business_logic/src/test/java/org/metplus/curriculum/domain/job/JobRepositoryFake.java
package org.metplus.curriculum.domain.job;

import org.metplus.curriculum.domain.Job;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class JobRepositoryFake implements JobRepository {

    private Map<String, Job> allJobs = new HashMap<>();
    public Exception saveThrowsException = null;

    @Override
    public Job save(Job job) {
        if(saveThrowsException != null)
            throw saveThrowsException;
        if(job.getJobId() == null || job.getJobId().length() == 0) {
            job.setJobId("" + allJobs.size());
            allJobs.put(job.getJobId(), job);
        } else {
            allJobs.put(job.getJobId(), job);
        }
        return job;
    }

    @Override
    public Job findById(String jobId) {
        return allJobs.get(jobId);
    }
}

File: business_logic/src/test/java/org/metplus/curriculum/domain/job/testDoubles/JobRepositoryFake.kt
package org.metplus.curriculum.domain.job.testDoubles

import org.metplus.curriculum.domain.job.Job
import org.metplus.curriculum.domain.job.JobRepository

class JobRepositoryFake : JobRepository {

    private val allJobs = HashMap<String, Job>()
    var saveThrowsException: Exception? = null

    override fun save(job: Job): Job {
        if (saveThrowsException != null)
            throw saveThrowsException as Exception

        val jobToSave = job.copy(
                jobId = if(job.jobId.isEmpty()) {
                    allJobs.size.toString()
                } else {
                    job.jobId
                }
        )

        allJobs.put(job.jobId, jobToSave)

        return job
    }

    override fun findById(jobId: String): Job? {
        return allJobs[jobId]
    }
}
